pkg/pillar/hypervisor/kvm.go

```diff
@@ -699,17 +708,19 @@ func (ctx KvmContext) Setup(status types.DomainStatus, config types.DomainConfig
 	domainName := status.DomainName
 	domainUUID := status.UUIDandVersion.UUID
 
-	swtpmCtrlSock, err := launchSwtpm(domainUUID.String(), swtpmTimeout)
-	if err != nil {
-		// let the vm start without TPM
-		logError("failed to launch swtpm for domain %s: %v", domainName, err)
-		swtpmCtrlSock = ""
-	}
-
-	// first lets build the domain config
-	if err := ctx.CreateDomConfig(domainName, config, status, diskStatusList,
-		aa, globalConfig, swtpmCtrlSock, file); err != nil {
-		return logError("failed to build domain config: %v", err)
+	// check if vTPM is enabled
+	if config.VirtualTPM {
```

Comment:
Maybe we can put just the var redifinitiaon under this if? =) I mean `swtpmCtrlSock`.
It's either set or `""`.
And leave
```
		if err := ctx.CreateDomConfig(domainName, config, status, diskStatusList,
			aa, globalConfig, swtpmCtrlSock, file); err != nil {
			return logError("failed to build domain config: %v", err)
		}
```
a common code )

Reply:
it was like that before, then I thought this looks cleaner, I will revert it back.